src/net/rpc.rs

```diff
@@ -3397,6 +3498,175 @@ mod test {
         );
     }
 
+    #[test]
```

Comment:
For each of these tests, can you please put some commenting that helps the reader understand what situation you are setting up and why we expect the given results?

I don't find it obvious.